static/app/views/alerts/rules/metric/ruleForm.tsx

```
@@ -1030,6 +1048,11 @@ class RuleFormContainer extends DeprecatedAsyncComponent<Props, State> {
 
     const {organization, project} = this.props;
     const {timeWindow, sensitivity, seasonality, thresholdType} = this.state;
+    // don't send the request if historical and current data have incorrect time windows
+    if (!this.TimeWindowsAreConsistent()) {
+      return;
```

Is silent failure ok here? The user has no way of differentiating this flow from the case where no anomalies were detected, right?

---

Also this addresses the symptom but not the cause. Do we know how we end up getting two different sets of data mixed up?

---

The root cause is the asynchronous requests that set state for current and historical data. Due to the async nature + the fact that this is a class component, there will be race conditions. We just cancel the anomaly detection request until both time series are synced.

---

@natemoo-re let me know if there's something else I should do instead.